addons/mail/static/src/models/notification_group/notification_group.js

```diff
@@ -97,15 +119,52 @@ function factory(dependencies) {
     }
 
     NotificationGroup.fields = {
-        date: attr(),
+        /**
+         * Determine the maximum (most distant future) date of all the message
```

Comment:
"most distant future" is a bit confusing in this context, because it will typically only have dates in the past.

I think "most recent date from all notification messages" like in the original code is enough?